front/src/pages/student-registration-page/index.js

```diff
@@ -10,71 +13,29 @@ const layout = {
   wrapperCol: { span: 16 },
 };
 
-const tailLayout = {
-  wrapperCol: {
-    offset: 9,
-    span: 16,
-  },
-};
-
 class StudentRegistrationPage extends Component {
   constructor(props) {
     super(props);
     this.handleSubmit = this.handleSubmit.bind(this);
-    this.validateEng = this.validateEng.bind(this);
-    this.validateUkr = this.validateUkr.bind(this);
-    this.validatePhoneNumber = this.validatePhoneNumber.bind(this);
-    this.confirmPassword = this.confirmPassword.bind(this);
+    this.renderStudentRegistrationForm = this.renderStudentRegistrationForm.bind(this);
   }
 
-  handleSubmit(data) {
-    data.groupToken = this.props.token;
+  componentDidUpdate() {
+    const {isRegistered, isFormSended, errorMessage, email } = this.props;
 
-    this.props.dispatch(studentRegistered(data));
-  }
-
-  async validateEng(_, value) {
-    const engRegexp = /^[A-Z]{1}([a-z]{1,})?$/;
-    if (engRegexp.test(value)) {
-      return Promise.resolve();
+    if (!isRegistered && isFormSended) {
+      showNotification('Sorry but you did not register', errorMessage, 'error');
     } else {
-      return Promise.reject('Not valid name or surname in English');
+      showNotification('Successfully registered', `User with provided email ${email} was registered`, 'success');
     }
   }
 
-  async validateUkr(_, value) {
-    const ukrRegexp = /^[А-ЯІЇЄ]{1}([а-яіїє]{1,})?$/;
-    const excludeRuLetters = /^[^ЫЭЪЁ]{1}([^ыэъё]{1,})?$/;
-
-    if (ukrRegexp.test(value) && excludeRuLetters.test(value)) {
-      return Promise.resolve();
-    } else {
-      return Promise.reject('Not valid name or surname in Ukrainian');
-    }
-  }
-
-  async validatePhoneNumber(_, value) {
-    const phoneNumberRegexp = /\+380\d{9}/;
-
-    if (phoneNumberRegexp.test(value)) {
-      return Promise.resolve();
-    } else {
-      return Promise.reject('Not valid phone number');
-    }
-  }
-
-  async confirmPassword({ getFieldValue }) {
-    return {
-      validator(rule, value) {
-        if (!value || getFieldValue('password') === value) {
-          return Promise.resolve();
-        }
-        return Promise.reject('The two passwords that you entered do not match!');
-      },
-    };
+  handleSubmit(data) {
+    data.groupToken = this.props.match.params.token;
+    this.props.dispatch(studentRegistered(data));
```

Comment:
Please, define `mapDispatchToProps` as a second argument of `connect` function and get functions from there

Reply:
Done